### PR TITLE
Add docker documentation for V14 and V15

### PR DESCRIPTION
**Pull request by nikolajlauridsen**
## Description

Updated the help output in the  `install-umbraco-with-templates.md` to include the new options, I also added an article about how you can run Umbraco locally with Docker using Docker Compose.  Note that this is only relevant for 14.3 and up, but I still added it to the V14 section as well.

## Type of suggestion

* [ ] Typo/grammar fix
* [X] Updated outdated content
* [X] New content
* [ ] Updates related to a new version
* [ ] Other

## Product & version (if relevant)

CMS v14 and v15

### CODE REVIEW

File: 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+﻿---
+description: "Running Umbraco on docker locally using docker compose"
+---
+
+# Running Umbraco in Docker using Docker Compose
+
+To aid in developing Umbraco with additional services, the templates can provide the requisite files to run Umbraco with a SQL Server database in Docker.
+This setup is designed to be used for local development, and not for production.
```
Comment (sofietoft) [inline 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md:8]:
```suggestion
To aid in developing Umbraco with additional services, the templates can provide the requisite files to run Umbraco with an SQL Server database in Docker. This setup is designed to be used for local development, and not for production.
```

File: 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+cd MyDockerProject
+```
+Next create your Umbraco project using the Umbraco Templates, and remember to use the `--add-docker` flag to include the Docker files.
+Conventionally this is named the same as the folder, but is not a requirement.
+
+```bash
+dotnet new umbraco -n MyDockerProject --add-docker
+```
+
+Now we need to add some additional files to make docker compose work. We can do this using the `umbraco-compose` template, passing the project name we specified earlier to the `-P` parameter:
```
Comment (sofietoft) [inline 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md:33]: Can we use steps here? 😇 

```suggestion
1. Create a folder to hold your project and enter that folder.

```bash
mkdir MyDockerProject
cd MyDockerProject
```

2. Create your Umbraco project using the Umbraco Templates, and remember to use the `--add-docker` flag to include the Docker files.

Conventionally this is named the same as the folder, but it is not a requirement.

```bash
dotnet new umbraco -n MyDockerProject --add-docker
```

Now we need to add some additional files to make docker compose work. We can do this using the `umbraco-compose` template, passing the project name we specified earlier to the `-P` parameter:
```

File: 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    * Typical project files
+    * DockerFile
+    * .dockerignore
+  * .env
+  * Database
+    * DockerFile
+    * healthcheck.sh
+    * setup.sql
+    * startup.sh
+  * docker-compose.yml
```
Comment (sofietoft) [inline 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md:54]: Use inline code for all file names.

```suggestion
* MyDockerProject
  * MyDockerProject
    * Typical project files
    * DockerFile
    * `.dockerignore`
  * `.env`
  * Database
    * DockerFile
    * `healthcheck.sh`
    * `setup.sql`
    * `startup.sh`
  * `docker-compose.yml`
```

File: 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+As you can see the project now includes docker files for both Umbraco itself and the SQL server database, as well as some additional scripts to launch and configure the database.
+
+It also includes a .env file with the password for the database.
```
Comment (sofietoft) [inline 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md:58]:
```suggestion
The project now includes docker files for both Umbraco and the SQL server database. 

It also includes additional scripts to launch and configure the database and a `.env` file with the database password.
```

File: 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+## Running
+
+To run the project run the `docker compose up` command from the root of the project, same folder as `docker-compose.yml`.
```
Comment (sofietoft) [inline 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md:62]:
```suggestion
To run the project use the `docker compose up` command in the root of the project files where the `docker-compose.yml` is.
```

File: 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+This command will build both the Umbraco and Sql Server images and launch them in the correct order, after a while the site will have booted and you can navigate to it in your browser like nomral on `http://localhost:44372/`
```
Comment (sofietoft) [inline 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md:64]:
```suggestion
This command will build both the Umbraco and SQL Server images and launch them in the correct order. When the site is booted, access it in your browser on `http://localhost:44372/`.
```

File: 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+### Useful commands
+
+There is some useful commands you can use to manage the docker containers:
```
Comment (sofietoft) [inline 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md:68]:
```suggestion
There are some useful commands you can use to manage the docker containers:
```

File: 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+* `docker compose down --volumes` This will delete your containers and the volumes they use, this is useful if you want to start from scratch.
+
+{% hint style="warning" %}
+This delete your database and all data in it, so be careful with this command.
+{% endhint %}
+
+* `docker compose up --build` This will rebuild the images and start the containers, this is useful if you have made changes to the project and want to see them reflected in the running site.
+* `docker compose watch` This will start the containers and watch the default models folder, this means that if the project uses source code models builder the images automatically rebuilds and restarts when you change the models.
```
Comment (sofietoft) [inline 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md:77]:
```suggestion
* `docker compose down --volumes`: Delete your containers and the volumes they use. This is useful if you want to start from scratch.

{% hint style="warning" %}
Be careful with this command, as it deletes your database and all data in it.
{% endhint %}

* `docker compose up --build`: Rebuild the images and start the containers. This is useful if you have made changes to the project and want to see them reflected on the running site.
* `docker compose watch`: Start the containers and watch the default models folder. This means that if the project uses a source-code models builder the images are automatically rebuilt and restarts when you change the models.
```

File: 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+## Details
+
+The docker compose file uses bind mounts for the following folders:
+
+* /wwwroot/media
+* /wwwroot/scripts
+* /wwwroot/css
+* /Views
+* /models
```
Comment (sofietoft) [inline 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md:87]:
```suggestion
* `/wwwroot/media`
* `/wwwroot/scripts`
* `/wwwroot/css`
* `/Views`
* `/models`
```

File: 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+This is bad for production, however, for local development this means that the files necessary for development are available from outside the container in your IDE allowing you to effectively develop even though the project is running in docker.
```
Comment (sofietoft) [inline 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md:89]:
```suggestion
This is not meant to be used in production.

For local development, however, this means that the files necessary for development are available from outside the container in your IDE. This allows development even though the project is running in docker.
```

File: 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+## Template options
+
+The `umbraco-compose` template has a few options that can be used to customize the setup:
+
+* -P or --project-name: The name of the project, this is required and used to set the correct paths in the docker-compose file.
+* -dbpw or --DatabasePasswor: Used to specify the database password, this is stored in the .env file, defaults to: Password1234
+* -p or --Port: Used to specify the port the site will run on, defaults to 44372
```
Comment (sofietoft) [inline 14/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally.md:97]:
```suggestion
* `-P` or `--project-name`: The name of the project. This is required and used to set the correct paths in the docker-compose file.
* `-dbpw` or `--DatabasePasswor`: Used to specify the database password. This is stored in the `.env` file and defaults to: `Password1234`.
* `-p` or `--Port`: Used to specify the port the site will run on. Defaults to `44372`.
```

**Review by sofietoft — CHANGES_REQUESTED** [review]
I've made some (okay, maybe a lot 🙈 ) suggestions to the new article.
Let me know if something doesn't look right.
I've only added comments to v14 - once they have been considered, the same changes should be added to v15 🙌 

Also, this is a new article.
It needs to be added to the SUMMARY file. Let me know if you prefer me doing this part 💪

**Comment by nikolajlauridsen** [comment]
Thanks 😄 I've applied the suggestions now, and good call on the list thing, didn't think of that 😄 

I didn't know about the summary file 😄 But yes that would be great if you could handle that, then I can see your commit for next time 😄

**Pull request merged**